src/clients/Deluge.ts

```diff
@@ -140,34 +168,38 @@ export default class Deluge implements TorrentClient {
 	 * parses the set-cookie header and updates stored value
 	 */
 	private handleResponseHeaders(headers: Headers) {
-		if (headers.has("Set-Cookie")) {
-			this.delugeCookie = headers.get("Set-Cookie").split(";")[0];
-		}
+		this.delugeCookie = headers.has("Set-Cookie")
+			? headers.get("Set-Cookie").split(";")[0]
+			: null;
 	}
 
 	/**
 	 * checks enabled plugins for "Label"
 	 * returns true if successful.
 	 */
 	private async labelEnabled() {
-		const enabledLabels = await this.call("core.get_enabled_plugins", []);
-		return enabledLabels?.result?.includes?.("Label");
+		const enabledPlugins = await this.call<string>(
+			"core.get_enabled_plugins",
+			[]
+		);
+		return enabledPlugins ? enabledPlugins.result.includes("Label") : false;
 	}
 
 	/**
 	 * if Label plugin is loaded, adds (if necessary)
 	 * and sets the label based on torrent hash.
 	 */
 	private async setLabel(infoHash: string, label: string): Promise<void> {
-		if (this.isLabelEnabled) {
-			const setResult = await this.call("label.set_torrent", [
-				infoHash,
-				label,
-			]);
-			if (setResult?.error?.code == DelugeErrorCode.RPC_FAIL) {
-				await this.call("label.add", [label]);
-				await this.call("label.set_torrent", [infoHash, label]);
-			}
+		const setResult = this.isLabelEnabled
+			? await this.call<void>("label.set_torrent", [infoHash, label])
+			: undefined;
+		if (!setResult) {
+			throw new Error(
+				"label.set_torrent: Client returned empty response"
+			);
```

Comment:
why is this needed? seems like it wasn't needed before

Reply:
I think I was handling if the Label plugin stack-traced and crashed. Honestly don't remember the specific case, but there was some condition that had me do it :P